### PR TITLE
Release Telemetry SDK Lite on Maven

### PR DESCRIPTION
**Pull request by electrostat**
Addresses: https://github.com/mapbox/mobile-telemetry/issues/272

Change pipeline system to release `Lite` and `Full` SDK on maven. Currently neither is working properly due to `Lite` refactor.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -93,6 +93,7 @@ jobs:
       IS_LOCAL_DEVELOPMENT: false
       OK_TO_RELEASE_CORE: true
       OK_TO_RELEASE_TELEM: true
+      OK_TO_RELEASE_LITE: false
```
Comment (andrlee) [inline .circleci/config.yml]: why not just release both at the same time? any potential issues you see?

File: Makefile [hunk 2]
```diff
@@ -20,6 +20,9 @@ publish-core:
 publish-telem:
 	export IS_LOCAL_DEVELOPMENT=false; ./gradlew :libtelemetry:uploadArchives
 
+publish-lite:
+    export IS_LOCAL_DEVELOPMENT=false; ./gradlew :libtelemetry:uploadArchives
```
Comment (andrlee) [inline Makefile]: does it look like it might be a dup of the above?

File: .circleci/config.yml (hunk 1 above)
Comment (electrostat) [reply .circleci/config.yml]: work is still in progress. Working on being able to release separately and then will be easy enough to combine if necessary.

File: Makefile (hunk 2 above)
Comment (electrostat) [reply Makefile]: see above

File: gradle/mvn-push-android.gradle [hunk 3]
```diff
@@ -97,6 +118,60 @@ afterEvaluate { project ->
...
+                        if (!VERSION_NAME.contains('LITE')) {
+                            String coreString = "com.mapbox.mapboxsdk:mapbox-android-core:" + core.version
+                            def coreArray = coreString.split(':')
+                            dependenciesArray.add(coreArray)
+
+                            String lifecycleString = dependenciesList.archLifecycleExtensions
+                            def lifecycleArray = lifecycleString.split(':')
+                            dependenciesArray.add(lifecycleArray)
+
+                            def viewModel = ["viewmodel", "android.arch.lifecycle"]
```
Comment (andrlee) [inline gradle/mvn-push-android.gradle]: why are we hardcoding all these? are we expected to hardcode every time we add/remove dependencies?

**Pull request merged**